File: stack/isValid.go
package main

import "fmt"

func isValid(s string) bool {
	pairs := map[rune]rune{
		'}': '{',
		']': '[',
		')': '(',
	}
	stack := make([]rune, 0)
	fmt.Println(stack, pairs)
	for _, item := range s {
		// fmt.Println("SSS", item)
		// fmt.Printf("%#T\n", item)
		pair, ok := pairs[item]
		if !ok {
			stack = append(stack, item)
			continue
		}
		if len(stack) == 0 {
			return false
		}
		if stack[len(stack)-1] != pair {
			return false
		}
		stack = stack[:len(stack)-1]
		// fmt.Println(stack, stack[len(stack)-1], pair)
		// if len(stack)==0 || stack[-1]!= pairs[item]

	}
	return len(stack) == 0
}
func main() {

	fmt.Println(isValid("()[]{}"))
}

File: stack/isValid.py
class Solution(object):
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """
        m = {
            ")":"(",
            "}":"{",
            "]":"[",
        }
        stack = []
        for item in s:
            if item not in m:
                stack.append(item)
                print(item, stack, "DDDD")
            else:
                if not stack or stack[-1]!= m[item]:

                    return False
                print(stack[-1], m[item], "ZZZZZ")
                # if stack[-1]!=m[item]
        print("DD", stack)
        return not stack
            # else:
            #     pass
s = "()"
# s = "()"
print(Solution().isValid(s))